File: 设计模式/设计模式/命令模式/单品/DanPinSwitchCommand.m
//
//  DanPinSwitchCommand.m
//  设计模式
//
//  Created by zachary spark on 2019/5/13.
//  Copyright © 2019 3g. All rights reserved.
//

#import "DanPinSwitchCommand.h"

@interface DanPinSwitchCommand()
@property(nonatomic,strong)DanPinRealization *DanPin;
@end

@implementation DanPinSwitchCommand

- (instancetype)init:(DanPinRealization *)DanPin{
    
    self = [super init];
    if(self){
        self.DanPin = DanPin;
    }
    return self;
    
}

- (void)execute{
    [self.DanPin ProductSwitchOff];
}

@end

File: 设计模式-Swift/设计模式-Swift/命令模式/基本实现/DanPinSwitchCommand.swift
//
//  DanPinSwitchCommand.swift
//  设计模式-Swift
//
//  Created by zachary spark on 2019/5/14.
//  Copyright © 2019 3g. All rights reserved.
//

import UIKit

class DanPinSwitchCommand: DanPinCommandProtocol {

    var DanPin:DanPinRealization
    
    init(_ DanPin:DanPinRealization) {
        self.DanPin = DanPin
    }
    
    func execute() {
        self.DanPin.ProductSwitchOff(NSNull.init())
    }
    
}
